# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1234813414 18000
#      Mon Feb 16 14:43:34 2009 -0500
# Node ID 8cb2a144f321e0f20fb430593a1f09dd96894603
# Parent  cf2163418140558f6be2dc9bc7d041f5ad955141
genprops.awk (emit_source): don't use + to concatenate strings

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-02-16  John W. Eaton  <jwe@octave.org>
+
+	* genprops.awk (emit_source): Don't use + to concatenate strings.
+
 2009-02-16  Jaroslav Hajek  <highegg@gmail.com>
 
 	* data.cc (NATIVE_REDUCTION): Use boolNDArray::any for native bool
 	summation, boolNDArray::sum for double-valued.
 
 2009-02-16  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ov-base-mat.cc (octave_base_matrix<MT>::assign (const octave_value_list&,
diff --git a/src/genprops.awk b/src/genprops.awk
--- a/src/genprops.awk
+++ b/src/genprops.awk
@@ -534,22 +534,22 @@ function emit_source ()
     for (i = 1; i <= idx; i++)
     {
       if (factory[i])
       {
         dval = defval[i];
         if (type[i] == "radio_property" || type[i] == "color_property")
       	{
       	  k = index (dval, "{");
-    	    dval = substr (dval, k+1);
-  	      l = index (dval, "}");
+	  dval = substr (dval, k+1);
+	  l = index (dval, "}");
       	  if (k > 0 && l > 0)
-	          dval = "\"" + substr (dval, 1, l-1) +  "\"";
-    	    else
-  	        dval = "octave_value ()";
+	    dval = "\"" substr (dval, 1, l-1) "\"";
+	  else
+	    dval = "octave_value ()";
       	}
 
         printf ("  m[\"%s\"] = %s%s;\n", name[i], dval,
                 (type[i] == "handle_property" || type[i] == "graphics_handle" ? ".as_octave_value ()" : "")) >> filename;
       }
     }
 
     printf ("\n  return m;\n}\n\n") >> filename;
